src/client/logging/levels.ts

```diff
@@ -9,14 +9,16 @@ import * as winston from 'winston';
 
 export enum LogLevel {
     // Larger numbers are higher priority.
+    Off = 50,
     Error = 40,
     Warn = 30,
     Info = 20,
     Debug = 10,
     Trace = 5
 }
-export type LogLevelName = 'ERROR' | 'WARNING' | 'INFORMATION' | 'DEBUG' | 'DEBUG-TRACE';
+export type LogLevelName = 'OFF' | 'ERROR' | 'WARNING' | 'INFORMATION' | 'DEBUG' | 'DEBUG-TRACE';
 const logLevelMap: { [K in LogLevel]: LogLevelName } = {
+    [LogLevel.Off]: 'OFF',
```

Comment:
"Off" is special and should never make it into the actual logger.  So it doesn't need to be converted to a level name.

```suggestion
export type LogLevelName = 'ERROR' | 'WARNING' | 'INFORMATION' | 'DEBUG' | 'DEBUG-TRACE';
const logLevelMap: { [K in LogLevel]: LogLevelName } = {
```